extensions/replication/tasks/ReplicateObject.js

```diff
@@ -132,33 +132,39 @@ class ReplicateObject extends BackbeatTask {
         }, cb);
     }
 
-    _publishReplicationStatus(sourceEntry, replicationStatus, params, cb) {
-        const { log, reason } = params;
+    _publishReplicationStatus(sourceEntry, replicationStatus, params) {
+        const { log, reason, kafkaEntry } = params;
         const updatedSourceEntry = replicationStatus === 'COMPLETED' ?
             sourceEntry.toCompletedEntry(this.site) :
             sourceEntry.toFailedEntry(this.site);
         updatedSourceEntry.setReplicationSiteDataStoreVersionId(this.site,
             sourceEntry.getReplicationSiteDataStoreVersionId(this.site));
         const kafkaEntries = [updatedSourceEntry.toKafkaEntry(this.site)];
-        return this.replicationStatusProducer.send(kafkaEntries, err => {
+        this.replicationStatusProducer.send(kafkaEntries, err => {
             if (err) {
                 log.error(
-                    'error publishing entry to replication status topic',
+                    'error in entry delivery to replication status topic',
                     { method: 'ReplicateObject._publishReplicationStatus',
                       topic: this.repConfig.replicationStatusTopic,
                       entry: updatedSourceEntry.getLogInfo(),
                       replicationStatus:
                       updatedSourceEntry.getReplicationStatus(),
                       error: err });
-                return cb(err);
             }
-            log.end().info('replication status published', {
-                topic: this.repConfig.replicationStatusTopic,
-                entry: updatedSourceEntry.getLogInfo(),
-                replicationStatus,
-                reason,
-            });
-            return cb();
+            // Commit whether there was an error or not to allow
+            // progress of the consumer, as best effort measure when
+            // there are errors. We can count on the sweeper to retry
+            // entries that failed to be published to kafka (because
+            // they will keep their PENDING status).
+            if (this.consumer) {
+                this.consumer.onEntryCommittable(kafkaEntry);
+            }
+        });
+        log.end().info('replication status published', {
```

Comment:
maybe only log this if an error didn't occur above https://github.com/scality/backbeat/pull/399/commits/63b260d0b2827d15501f984815fd0ace8fae6b76#diff-e5fe2d1a77d806ec6632afc36abf102fR145

Reply:
This is logged in the synchronous context, so at this point we can only assert that we have published but did not get the feedback yet.